CRAMv3.tex

```diff
@@ -458,10 +459,11 @@ \section{\textbf{Container structure}}
 \hline
 ltf8 & bases & number of read bases\tabularnewline
 \hline
-itf8 & number of blocks & the number of blocks\tabularnewline
+itf8 & number of blocks & the total number of blocks in this container\tabularnewline
 \hline
-itf8[ ] & landmarks & Each integer value of this array is a byte offset into the 
-blocks byte array. Landmarks are used for random access indexing.\tabularnewline
+itf8[ ] & landmarks & the locations of slices in this container as byte offsets from the end of 
+this container header, used for random access indexing.
+The landmark count must equal the slice count.\tabularnewline
```

Comment:
Wouldn't this definition imply that the first landmark entry would always be 0, since the first slice header would always be immediately after the container header ? That doesn't seem to match the cram_dump output that @jkbonfield put in the comment in https://github.com/samtools/hts-specs/issues/396, nor what I see in htsjdk code:
```
final int containerHeaderSize = landmarks[0];
final int containerTotalByteSize = containerHeaderSize + containerByteSize;
```
Not sure which is incorrect, but the code seems to count the header size in the landmark offset, as if it were an offset from the beginning of the header/container, not the end of the header ?

Reply:
landmarks[0] = compression header length.  We could add a note here saying so.

I can make a quick PR to name this value more precisely in htsjdk.

Reply:
Also, these states are equivalent for containers:
* no landmarks = no slices = no compression header

Reply:
Ah I see - I wasn't properly distinguishing compression header from container header. I was thinking of them as one and a single header, but that doesn't match the spec treatment. I think your suggestion of noting that here is a good one, and would make that explicit.

Reply:
Thanks for the new picture.  Yes, it has helped to clarify this confusion.

I updated the text here in a new commit.

Reply:
I got muddled with which this PR was (too many related ones), sorry, but yes the picture elsewhere clarifies things there and your explicit comment regarding landmark[0] being size of the compression header helps a lot.